common/spec/dependabot/file_fetchers/base_spec.rb

```diff
@@ -1518,5 +1518,41 @@ def fetch_files
         end
       end
     end
+
+    describe "#clone_repo_contents" do
+      subject(:clone_repo_contents) do
+        file_fetcher_instance.clone_repo_contents
+      end
+
+      let(:repo) do
+        "dependabot-fixtures/go-modules-app"
+      end
+
+      it "clones the repo" do
+        clone_repo_contents
+        expect(`ls #{repo_contents_path}`).to include("README")
+      end
+
+      context "with a branch name including bash command" do
+        let(:branch) do
+          "\"$(time)\""
+        end
+
+        it "clones the repo" do
+          clone_repo_contents
+          expect(`ls #{repo_contents_path}`).to include("time.md")
```

Comment:
Xref https://github.com/dependabot-fixtures/go-modules-app/tree/%22%24(time)%22
I wondered what was going on here before I looked up the fixture.

I thought it might be slick to verify the command is escaped locally, e.g. `let(:branch) { "$(touch #{repo_contents_path}/time.md)" }`, then verify the file is _not_ there.
But I like how this verifies the clone not only blocks the command from running, but clones! Noice!

Reply:
Updated the spec copy a bit to make this clearer